apps/admin-server/src/pages/projects/[project]/areas/index.tsx

```diff
@@ -14,6 +14,43 @@ export default function ProjectAreas() {
   const { project } = router.query;
   const { data, removeArea } = useArea(project as string);
 
+  const [filterData, setFilterData] = useState([]);
+
+  useEffect(() => {
+    setFilterData(data);
+    console.log(data)
+  }, [data])
+
+  const sortFunctions = {
+    'id': (a: any, b: any) => b.id - a.id,
+    'name': (a: any, b: any) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()),
+  };
+
+  const sortTable = (sortType: string, el: React.MouseEvent<HTMLElement, MouseEvent>) => {
+    const sortFunction = sortFunctions[sortType as keyof typeof sortFunctions];
+    if (!sortFunction) {
+      console.error(`Invalid sortType: ${sortType}`);
+      return;
+    }
+
+    const filterButtons = document.querySelectorAll('.filter-button');
+    filterButtons.forEach(button => button.classList.remove('font-bold'));
+    filterButtons.forEach(button => button.classList.remove('text-black'));
+
+    el.currentTarget.classList.toggle('--up');
+    el.currentTarget.classList.add('font-bold');
+    el.currentTarget.classList.add('text-black');
+
+    const direction = el.currentTarget.classList.contains('--up') ? 'up' : 'down';
+
+    const sortedWidgets = [...filterData].sort((a: any, b: any) => {
+      const result = sortFunction(a, b);
+      return direction === 'up' ? result : -result;
+    });
+
+    setFilterData(sortedWidgets);
+  };
```

Comment:
Ik zie dit stuk code (met kleine wijzigingen) vaker voorkomen in deze PR. Zou je dit kunnen refactoren naar een los bestand?